taichi/transforms/compile_to_offloads.cpp

```diff
@@ -82,12 +82,12 @@ void compile_to_offloads(IRNode *ir,
   irpass::offload(ir);
   print("Offloaded");
 
-  irpass::full_simplify(ir, config);
-  print("Simplified III");
-
   irpass::demote_atomics(ir);
   print("Atomics demoted");
 
+  irpass::full_simplify(ir, config);
+  print("Simplified III");
+
```

Comment:
I moved `simplify III` here rather than added a `simplify IV` pass because this pass takes some time.

I compiled some kernels several times, and if there was only one `simplify III` pass here, it took 34.401/34.325/34.809s;
if there were both `simplify III` and `simplify IV`, it took 35.993/34.104/35.241s.

The speedup is more than 1%.

On the other hand, moving it after `demote_atomics` always reduces the number of statements, if not keeping the number of statements the same.

Reply:
Sounds good!